components/browser/state/src/test/java/mozilla/components/browser/state/reducer/LocaleStateReducerTest.kt

```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.reducer
+
+import mozilla.components.browser.state.action.LocaleAction
+import mozilla.components.browser.state.state.BrowserState
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Test
+import java.util.Locale
+
+class LocaleStateReducerTest {
+
+    @Test
+    fun `WHEN updating locale action is called THEN the reducer updates locale state`() {
+        val reducer = LocaleStateReducer
+        val state = BrowserState()
+        val locale = Locale("es")
+        val action = LocaleAction.UpdateLocaleAction(locale)
+
+        assertNull(state.locale)
+
+        val result = reducer.reduce(state, action)
+
+        assertEquals(result.locale, locale)
+    }
+
+    @Test
+    fun `WHEN restoring locale action is called THEN the reducer persists state`() {
```

Comment:
The reducer doesn't do anything with this action (the middleware is), so the test name needs an update here e.g. THEN the state is not change...or similar.